### PR TITLE
Add function-by-function testing

### PR DESCRIPTION
**Pull request by YagoGG**
This PR adds testing with [Mocha](http://mochajs.org/) (as the testing framework) and [Nock](https://github.com/pgte/nock) (for mocking GitHub's Developer API).

It works by checking that all the main functions, now exported to `logic.js`, behave properly (i.e. no exceptions thrown), one by one.

The testing procedure has been slightly documented and included as a script in the `package.json` file.

### CODE REVIEW

File: test/test.js [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+
+function responseLog(response) {
+  console.log('   RESPONSE SENT: ' +  response);
+}
+
+var ghUser = nock('https://api.github.com')
+              .get('/user')
+              .reply(200, {
+                login: 'testuser',
+                id: 1234567,
```
Comment (roonyh) [inline test/test.js:14]: you can put only the parameters we need. and maybe some important few other parameters. like id and avatar_url.

File: test/test.js [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+                followers: 0,
+                following: 0,
+                created_at: '2016-00-00T00:13:37Z',
+                updated_at: '2016-03-14T00:15:09Z'
+              });
+
+describe('welcome()', function() {
+  it('Should be able to authenticate with GitHub', function(done) {
+    logic.welcome('token', function(err, user) {
+      done(err);
```
Comment (roonyh) [inline test/test.js:27]: so you can use assert and test if the function behaves as expected.

`assert(user=='testuser')`

there are some libraries that you can use as well. to write tests more elegantly than using basic 'assert'. see http://chaijs.com/

**Comment by roonyh** [comment]
The server code is very nice now!. Good job on tests. See my comments

**Comment by YagoGG** [comment]
Thank you @roonyh! Working on them...

**Comment by YagoGG** [comment]
2a2c5a9 reduces the content sent in the body of the _fake_ responses generated by Nock.
7c8cfa2 adds some more tests. This should be enough to check that the data gathered from GitHub is properly read. Used `assert()` for that ;)

About [Chai](http://chaijs.com/)... I don't really like it. It's okay that that programming languages use words for the _common languages_ to explain what they do, but I don't expect to program like I talk. It just doesn't feel right to me, and think that may lead to syntax problems while developing, if you're not careful.  
Of course, this is a pesonal opinion. If you prefer using Chai instead of assert, just let me know and will make another commit.

**Comment by roonyh** [comment]
This looks great now! 

It's fine you don't like chai :)

When you work with more complex tests you might start to like it better. Assert is enough for us for now.

You can submit the task now.

**Comment by YagoGG** [comment]
Came across this while doing some open PR cleanup.

I'm closing this since the project isn't active anymore. Thanks a lot for your help, @roonyh!

**Pull request closed without merging**